lib/statsample-glm/glm/base.rb

```diff
@@ -144,7 +144,7 @@ def add_constant_vector x=nil
         @data_set.add_vector :constant, [@opts[:constant]]*@data_set.nrows
 
         unless x.nil?
-          @data_set.add_vector :constant, [1]*@data_set.nrows
+          @data_set.add_vector :constant, [x]*@data_set.nrows
```

Comment:
Well, how it works now is that if you call `add_constant_vector` (without an argument), then it is going to add a constant vector with value `@opts[:constant]` (this is what we want most of the time). But if you call `add_constant_vector(x)`, then it is going to add a constant vector with value of `x`.

Maybe I should change the code like this to make it more clear?
```Ruby
def add_constant_vector x=nil
  if x.nil?
    @data_set.add_vector :constant, [@opts[:constant]]*@data_set.nrows
  else
    @data_set.add_vector :constant, [x]*@data_set.nrows
  end
end
```

Or maybe have it simply like this, and call `add_constant_vector @opts[:constant]` (instead of `add_constant_vector` without argument)?
```Ruby
def add_constant_vector x=1
  @data_set.add_vector :constant, [x]*@data_set.nrows
end
```

Which do you think is the better alternative? (I tend to the second one now)